### PR TITLE
Make firewalld operations idempotent

### PR DESCRIPTION
**Pull request by sanatsathaye**
I haven't worked much with `ansible.posix.firewalld` before, so let me know if there's something which needs changing. Think I've translated it over correctly though.

### CODE REVIEW

**Review by codyro — CHANGES_REQUESTED** [review]
We should add a `requirements.yml` file with the  `ansible.posix` collection as well.

```yaml
---
collections:
  - name: ansible.posix
```

File: lemmy-almalinux.yml [hunk 1]
```diff
@@ -44,6 +44,11 @@
       ansible.builtin.systemd:
         name: nginx
         state: reloaded
+
+    - name: Reload firewalld
```
[the remark this answers is not in the document]
Comment (sanatsathaye) [reply lemmy-almalinux.yml:48]: Hmm, so basically with `immediate: true` we don't need to an explicit reload at all? The module will do it on its own?

[the remark this answers is not in the document]
Comment (codyro) [reply lemmy-almalinux.yml:48]: Yes, that is correct. 

It might help to understand what `permanent` means in the context of `firewallD`. 

This command will allow `http` traffic in the `public` zone and take effect *immediately*. The rule will *not* be persistent across reboots/reloads.

```shell
firewall-cmd --zone=public --add-service=http
````
Comparatively, the changes are saved but not applied when you use the `--permanent` flag (which is what the modules `permanent: true` refers to). As such, you'd usually pair it with `firewall-cmd --reload`:

```shell
firewall-cmd --zone=public --add-service=http --permanent # Not applied yet
firewall-cmd --reload # Now it's applied
```
So, in short, when you pair `immediate: true` with `permanent: true`, it effectively does the above, making the reload redundant. 

There is a caveat with this when handling zones, requiring you to reload firewalld explicitly if you want to use it immediately [1]. However, it's unnecessary since we use the default `public` zone.

[1] https://docs.ansible.com/ansible/latest/collections/ansible/posix/firewalld_module.html#notes

**Comment by sanatsathaye** [comment]
Sorry for the delay I've been very busy IRL with work and exams, I'll resume working on this some time next week

**Comment by codyro** [comment]
No rush! Thank you for your time on improving these playbooks and the Lemmy ecosystem as a whole!

File: lemmy-almalinux.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanatsathaye) [reply lemmy-almalinux.yml:48]: Hmm okay, I'll keep the handler for now then but will remove the notify

**Comment by sanatsathaye** [comment]
No problem, need a good reddit alternative 😄
Btw I suppose I'll have to put instructions to run `requirements.yml` in the readme somewhere?

**Comment by codyro** [comment]
> I suppose I'll have to put instructions to run `requirements.yml` in the readme somewhere?

If you don't mind, that'd be fantastic. While doing that, could you add that these playbooks require `Ansible >= 2.10`? We ran into an issue with `notify` not being supported in `blocks` in older versions (#177).

Thank you!

**Review by codyro — APPROVED** [review]

**Comment by sanatsathaye** [comment]
> While doing that, could you add that these playbooks require `Ansible >= 2.10`? We ran into an issue with `notify` not being supported in `blocks` in older versions (#177).

I believe it's possible to use `ansible` itself to check which version is running the playbook using `{{ ansible_version }}` -- I can do another PR to add that into `lemmy.yml` and `lemmy-almalinux.yml`, and add the requirement in the readme as well if that's alright. Don't think it should be needed for the uninstall playbook.

Btw, should we be tying down the collections in the `requirements.yml` to any specific version number? Just realised I'll need to add `community.docker` as well as that's part of the Ubuntu install playbook, so I can point that as well as `ansible.posix` to the current latest version of both if that is required. If not, then I'll just add as is.

**Review by dessalines — APPROVED** [review]
